src/Umbraco.Web/HttpContextAccessorExtensions.cs

```diff
@@ -0,0 +1,17 @@
+using System.IO;
+using System.Web;
+
+namespace Umbraco.Web
+{
+    public static class HttpContextAccessorExtensions
+    {
+        public static HttpContextBase GetRequiredHttpContext(this IHttpContextAccessor httpContextAccessor)
+        {
+            var httpContext = httpContextAccessor.HttpContext;
+
+            if(httpContext is null) throw new IOException("HttpContext is null");
```

Comment:
How come it's IOException?

Reply:
Dooh'  Should have been InvalidOperationException..  I must have chosen the wrong one, when the suggestions came..  Do you agree on InvalidOperationException? - Then I fix in dev

Reply:
Yep sounds good